image_writers_guide/guidelines.adoc

```diff
@@ -9,4 +9,135 @@
+
+=== Place instructions in proper order
+Docker reads the Dockerfile and runs the instructions from top to bottom. Every successfully executed instruction creates a layer which can be reused next we build this (or other image). It is very important to place on top of your Dockerfile instructions that will change rarely.
+
+This will make sure the next builds of the same image will be very fast because the cache will not be invalidated by upper layer changes.
+
+For example, if you are working on a Dockerfile that contains an `ADD` command to install a file you are iteratin on, and a `RUN` command to yum install a package, it's best to put the `ADD` command last, such as:
+
+		FROM foo
+		RUN yum -y install mypackage && yum clean all -y
+		ADD myfile /test/myfile
+
+This way each time you edit `myfile` and rerun `docker build` the system will reuse the cached layer for the yum command and only have to generate the new layer for the `ADD` operation.
+
+If instead you wrote the Dockerfile as:
+
+		FROM foo
+		ADD myfile /test/myfile
+		RUN yum -y install mypackage && yum clean all -y
+
+Then each time you changed `myfile` and reran `docker build`, the `ADD` operation would invalidate the `RUN` layer cache so the yum operation would need to be rerun as well.
+
+=== Use EXPOSE to mark important ports
+See http://www.projectatomic.io/docs/docker-image-author-guidance/, subsection _Always `EXPOSE` Important Ports_
+
+=== Set environment variables with ENV
+It’s a good idea to set environment variables with the `ENV` instruction. One example is to set version of your project. This will make it easy for people find the version without even looking at the Dockerfile.  Another example of an environment variable worth setting is when to advertise some path on the system that could be used by some process, like `JAVA_HOME`.
+
+=== Do not set default passwords
+It is not a good idea to set default passwords. Many people will extend the image and forget to remove or change the default password. This will lead to security issues where in production there will be a user with a well-known passwords. Passwords should be configurable via environment variable instead (See anchor:configuration[])
+
+If you do choose to set a default password, ensure that an appropriate warning message is displayed when the container is started.  The message should inform the user of the value of the default password and explaining how to change it (e.g. what environment variable to set).
+
+=== Do not start (or install) SSHD
+Running SSHD should be avoided.  For accessing running containers you can use `docker exec` locally, or the OpenShift tooling which allows execution of arbitrary commands in running images.  Installing and running SSHD in your image opens up additional vectors for attack and requirements for security patching.
```

Comment:
you mean like the second sentence says? " For accessing running containers you can use `docker exec` locally"  :)